cas/service.c

```diff
@@ -1923,6 +1926,20 @@ static int cas_id_check(const struct m0_cas_id *cid)
 		layout = &cid->ci_layout;
 		if (layout->dl_type != DIX_LTYPE_DESCR)
 			rc = M0_ERR(-EPROTO);
+		if (fom != NULL) {
+			svc = M0_AMB(svc, fom->cf_fom.fo_service, c_service);
+			device_id = m0_dix_fid_cctg_device_id(&cid->ci_fid);
+			if (svc->c_sdev_id != INVALID_CAS_SDEV_ID &&
+			    svc->c_sdev_id != device_id &&
+			    cas_type(&fom->cf_fom) != CT_META) {
+				M0_ASSERT_INFO(svc->c_sdev_id == device_id,
```

Comment:
We will not see this message if the asserts are disabled, right? If so then I propose to simply log the error here and return the -EPROTO since this error message is very important for the testing and production usage, can use M0_ERR_INFO(). Also please check the indentation of strings.